instrumentation/opensearch/opensearch-rest-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opensearch/rest/v1_0/RestClientInstrumentation.java

```diff
@@ -0,0 +1,134 @@
+
+import static io.opentelemetry.javaagent.bootstrap.Java8BytecodeBridge.currentContext;
+import static net.bytebuddy.matcher.ElementMatchers.isMethod;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer;
+import io.opentelemetry.javaagent.instrumentation.opensearch.rest.OpenSearchRestRequest;
+import io.opentelemetry.javaagent.instrumentation.opensearch.rest.RestResponseListener;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.opensearch.client.Request;
+import org.opensearch.client.Response;
+import org.opensearch.client.ResponseListener;
+
+public class RestClientInstrumentation implements TypeInstrumentation {
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return named("org.opensearch.client.RestClient");
+  }
+
+  @Override
+  public void transform(TypeTransformer transformer) {
+    transformer.applyAdviceToMethod(
+        isMethod()
+            .and(named("performRequest"))
+            .and(takesArguments(1))
+            .and(takesArgument(0, named("org.opensearch.client.Request"))),
+        this.getClass().getName() + "$PerformRequestAdvice");
+    transformer.applyAdviceToMethod(
+        isMethod()
+            .and(named("performRequestAsync"))
+            .and(takesArguments(2))
+            .and(takesArgument(0, named("org.opensearch.client.Request")))
+            .and(takesArgument(1, named("org.opensearch.client.ResponseListener"))),
+        this.getClass().getName() + "$PerformRequestAsyncAdvice");
+  }
+
+  @SuppressWarnings("unused")
+  public static class PerformRequestAdvice {
+
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static void onEnter(
+        @Advice.Argument(0) Request request,
+        @Advice.Local("otelRequest") OpenSearchRestRequest otelRequest,
+        @Advice.Local("otelContext") Context context,
+        @Advice.Local("otelScope") Scope scope) {
+
+      Context parentContext = currentContext();
+      otelRequest = OpenSearchRestRequest.create(request.getMethod(), request.getEndpoint());
+      if (!OpenSearchRestSingletons.instrumenter().shouldStart(parentContext, otelRequest)) {
+        return;
+      }
+
+      context = OpenSearchRestSingletons.instrumenter().start(parentContext, otelRequest);
+      scope = context.makeCurrent();
+    }
+
+    @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
+    public static void stopSpan(
+        @Advice.Return(readOnly = false) Response response,
```

Comment:
this isn't needed since you aren't reassigning the return value below
```suggestion
        @Advice.Return Response response,
```

Reply:
Thanks for this, I corrected it